### PR TITLE
Adding and removing ssrcs from participant2sources

### PR DESCRIPTION
**Pull request by Kenny-House**
This handles source-adds and removes from participants such that the correct ssrcs are sent in invites to new participants

I'm not a seasoned lua developer by any means, so there are probably ways to optimize this code.

Handles https://github.com/otalk/mod_muc_focus/issues/51

### CODE REVIEW

**Comment by fippo** [comment]
paging @legastero

File: mod_muc_focus.lua [hunk 1]
```diff
@@ -1062,6 +1060,43 @@ module:hook("iq/bare", function (event)
...
+                end
+
+                local sources_of_name = participant2sources[room.jid][sender.nick][name] or {}
+                for i=#sources_of_name,1,-1 do
+                  local source = participant2sources[room.jid][sender.nick][name][i]
+                  if should_remove[source.attr.ssrc] then
+                    table.remove(participant2sources[room.jid][sender.nick][name], i)
+                    module:log("debug", "Removed source with ssrc %s from participant2sources, new size: %d", source.attr.ssrc, #sources_of_name)
+                  end --End if should_remove
+                end --End iterating through array
```
Comment (legastero) [inline mod_muc_focus.lua]: This feels like a potential resizing-array-while-iterating gotcha if there were multiple sources removed. Constructing a new array with the correct information and then swapping it in one step would be better.

**Comment by Kenny-House** [comment]
Good call, I've updated the source-remove part of the code to create a new array for each `name`, add any not-removed sources to it, and then assign that to `participant2sources[room.jid][sender.nick][name]`

**Comment by legastero** [comment]
LGTM, thanks!

**Pull request merged**